addon/m3-tracked-array.js

```diff
@@ -25,20 +25,35 @@ export default class M3TrackedArray extends ArrayProxy {
   }
 
   replaceContent(idx, removeAmt, newItems) {
-    newItems = newItems.map((item, idx) => {
+    // Update childModelDatas array
+    // mapping to array of nested models
+    const childModelDatas = this._model._internalModel._modelData.__childModelDatas;
+    const key = this._key;
+    if (childModelDatas && childModelDatas[key] && Array.isArray(childModelDatas[key])) {
+      const newItemsInChildModelData = new Array(newItems.length);
+      Array.prototype.splice.apply(
+        childModelDatas[key],
+        [idx, removeAmt].concat(newItemsInChildModelData)
+      );
+    }
+
+    let startIdx = idx;
+    newItems = newItems.map(item => {
       if (isResolvedValue(item)) {
         associateRecordWithRecordArray(item, this);
         return item;
       }
 
+      const index = startIdx;
+      startIdx++;
```

Comment:
i would personally see this more clearly if the index were computed as the sum of `idx` and the index argument to `map`

Reply:
updated the code to compute the index.